bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/SecureHttpClientFactory.java

```diff
@@ -0,0 +1,278 @@
+    }
+
+    @Override
+    public CommonHttpClient getCommonHttpClient() {
+        initialize();
+        logger.debug("shared httpClient requested");
+        return commonHttpClientFacade;
+    }
+
+    private void getConfigParameters(Map<String, Object> parameters) {
+        minThreadsShared = getConfigParameter(parameters, CONFIG_MIN_THREADS_SHARED, 10);
+        maxThreadsShared = getConfigParameter(parameters, CONFIG_MAX_THREADS_SHARED, 40);
+        keepAliveTimeoutShared = getConfigParameter(parameters, CONFIG_KEEP_ALIVE_SHARED, 60);
+        minThreadsCustom = getConfigParameter(parameters, CONFIG_MIN_THREADS_CUSTOM, 5);
+        maxThreadsCustom = getConfigParameter(parameters, CONFIG_MAX_THREADS_CUSTOM, 10);
+        keepAliveTimeoutCustom = getConfigParameter(parameters, CONFIG_KEEP_ALIVE_CUSTOM, 60);
+    }
+
+    private int getConfigParameter(Map<String, Object> parameters, String parameter, int defaultValue) {
+        if (parameters == null) {
+            return defaultValue;
+        }
+        Object value = parameters.get(parameter);
+        if (value == null) {
+            return defaultValue;
+        }
+        if (value instanceof Integer) {
+            return (Integer) value;
+        }
+        if (value instanceof String) {
+            try {
+                return Integer.parseInt((String) value);
+            } catch (NumberFormatException e) {
+                logger.warn("ignoring invalid value {} for parameter {}", value, parameter);
+                return defaultValue;
+            }
+        } else {
+            logger.warn("ignoring invalid type {} for parameter {}", value.getClass().getName(), parameter);
+            return defaultValue;
+        }
+
+    }
+
+    private synchronized void initialize() {
+        if (threadPool == null || commonHttpClient == null) {
+            try {
+                AccessController.doPrivileged(new PrivilegedExceptionAction<Void>() {
+                    @Override
+                    public Void run() {
+                        if (threadPool == null) {
+                            threadPool = createThreadPool("common", minThreadsShared, maxThreadsShared,
+                                    keepAliveTimeoutShared);
+                        }
+
+                        if (commonHttpClient == null) {
+                            commonHttpClient = createHttpClientInternal("common", null);
+                            commonHttpClientFacade = new HttpClientDelegate(commonHttpClient);
+                            logger.info("jetty shared http client created");
+                        }
+
+                        return null;
+                    }
+                });
+            } catch (PrivilegedActionException e) {
+                Throwable cause = e.getCause();
+                if (cause instanceof RuntimeException) {
+                    throw (RuntimeException) cause;
+                } else {
+                    throw new RuntimeException(cause);
```

Comment:
The intention here is to rethrow any RuntimeException (or subclasses), which were thrown in the run-method above.
The else-block is only needed because the cause of a PrivilegedActionException is a generic Throwable, so it cannot be rethrown directly. In this case it will never be a checked exception, because the run-method in the action does not declare any exceptions. So the else-part will never be executed. No need for a custom exception.

Reply:
It's a commonly acknowledged anti-pattern. 

Your arguments cannot convince me to be a valid exception in any of the cases. 

Some more background can be found at https://wiki.sei.cmu.edu/confluence/display/java/ERR07-J.+Do+not+throw+RuntimeException%2C+Exception%2C+or+Throwable and many other places.